src/librbd/api/Trash.cc

```diff
@@ -187,7 +214,7 @@ int Trash<I>::move(librados::IoCtx &io_ctx, rbd_trash_image_source_t source,
   r = ctx.wait();
   ictx->state->close();
```

Comment:
You can't close `ictx` here any more. It needs to be moved under `if (r == -ENOENT)` conditional block above. And then you can't use `ictx->id` and `ictx->name` above.

Reply:
@trociny I'm not sure I understand this. Why can't I close? Also, the preceding if condition checks for `if (r != -ENOENT)`